pkg/auth/userregistry/identitymapper/mapper.go

```diff
@@ -0,0 +1,71 @@
+package identitymapper
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/user"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	identityregistry "github.com/openshift/origin/pkg/user/registry/identity"
+	userregistry "github.com/openshift/origin/pkg/user/registry/user"
+	mappingregistry "github.com/openshift/origin/pkg/user/registry/useridentitymapping"
+)
+
+type UserStrategyType string
+
+const (
+	// StrategyTypeLookup does not provision a new identity or user, it only allows identities already associated with users
+	StrategyTypeLookup UserStrategyType = "lookup"
+
+	// StrategyTypeClaim associates a new identity with a user with the identity's preferred username
+	// if no other identities are already associated with the user
+	StrategyTypeClaim UserStrategyType = "claim"
```

Comment:
> claim sounded like the strategy that took over the user no matter what. maybe adopt?

These are all problematic.  I don't think any short identifier is going to be sufficient to describe the behavior.